exp/api/v1beta1/azuremanagedmachinepool_webhook.go

```diff
@@ -267,6 +267,10 @@ func (m *AzureManagedMachinePool) validateLastSystemNodePool(cli client.Client)
 		return nil
 	}
 
+	if ownerCluster.Spec.Paused {
```

Comment:
I think I'd prefer some other "clusterctl move"-specific indication that we're O.K. to delete the system node pool. Does a move operation result in an annotation or other metadata that we can detect?

Also, we should probably just implement this in the `(m *AzureManagedMachinePool) ValidateDelete` method itself.

Reply:
Currently, I see there is no annotation is added in the move operation.
Move client library ref: https://github.com/kubernetes-sigs/cluster-api/blob/5eaf9e3c0e93d93b75e57675a7e9fb560d15ae37/cmd/clusterctl/client/cluster/mover.go#L334

Reply:
If we want to implement this in ValidateDelete method, we need to get the cluster object which is already done in `func (m *AzureManagedMachinePool) validateLastSystemNodePool` method. I think, some code will be repeated then.

Reply:
@Jont828 @fabriziopandini does either of you know if there is any marker that indicates a delete is initiated as part of a move operation more specific than the pause annotation?

Reply:
currently, we don't add any markers to resources before the move.
cluster being paused is the only proxy information, but it is not 100% reliable because a cluster can be paused for any reason.

note: if required, adding a "move" marker on one or more move-hierarchies root (cluster, cluster class, crs) is something that makes sense; adding a "move" marker on all resources is a little bit more complex...

Reply:
IMO this is better than nothing, maybe we can open an issue in CAPI to add a more granular "move" annotation to the owner Cluster, but in the meantime this is a bug that prevents moving AKS clusters so we might want to proceed with this approach even if it's not perfect

Reply:
I think I agree with @CecileRobertMichon

There are going to be edge cases with this change but I think they are sufficiently "weird" that preventing them isn't more important than addressing this broken core capi functionality for AKS.

My only feedback at this point would be a unit test for this (I think `TestAzureManagedMachinePool_ValidateCreate` is a good copy/paste candidate for scaffolding a new UT for this func.

Thank you @pkbhowmick!

Reply:
So, should I only add a new UT for this func?

Reply:
Correct, in order to UT this new func, you'll need a new func (for example a new UT func `TestValidateLastSystemNodePool` based on existing pattern in `TestAzureManagedMachinePool_ValidateCreate`).

You will be doing the project an extra credit favor by adding coverage for an existing func that we don't have coverage for, thank you!

Reply:
Got it! Thank you!

Reply:
Hi @jackfrancis!
The function has dependency on client.Client interface. Is there any existing implementation of fake client to use? if not then I can implement one!

Reply:
I think `TestAzureManagedMachinePool_ValidateCreate` does this already